libs/RobotKit/tests/RobotController_test_stateAutonomousActivities.cpp

```diff
@@ -124,26 +125,100 @@ TEST_F(RobotControllerTest, stateGameActivityDisplayTagsChosen)
 
 	EXPECT_CALL(mock_videokit, displayImage).Times(1);
 
+	activitykit.start(MagicCard::number_10);
+
+	EXPECT_CALL(mock_videokit, displayImage).Times(0);
+
 	rc.onMagicCardAvailable(MagicCard::number_10);
 
 	EXPECT_TRUE(rc.state_machine.is(lksm::state::autonomous_activities));
 }
 
-TEST_F(RobotControllerTest, stateGameMagicCardAvailableActivityAlreadyStarted)
+// ! TODO: Refactor with composite SM & CoreTimer mock
+TEST_F(RobotControllerTest, stateGameActivityStartedBackToMenu)
```

Comment:
According to sonarcloud, 1 condition is missing 

https://github.com/leka/LekaOS/blob/a587177bba74b36bc3f910352174369afe67bdc2/libs/RobotKit/include/RobotController.h#L328

I suppose

```suggestion
// ! TODO: Refactor with composite SM & CoreTimer mock
TEST_F(RobotControllerTest, stateGameActivityStartedBackToMenuDelayNotOver)
{
	auto maximal_delay_before_over = 1s;

	rc.state_machine.set_current_states(lksm::state::autonomous_activities);

	auto set_activitykit_is_playing = [this] {
		const std::unordered_map<MagicCard, interface::Activity *> activities {{MagicCard::number_10, &display_tag}};
		activitykit.registerActivities(activities);
		EXPECT_CALL(mock_videokit, displayImage).Times(1);
		activitykit.start(MagicCard::number_10);
	};
	set_activitykit_is_playing();

	EXPECT_CALL(mock_videokit, displayImage).Times(0);

	spy_kernel_addElapsedTimeToTickCount(maximal_delay_before_over);
	rc.onMagicCardAvailable(MagicCard::dice_roll);

	EXPECT_TRUE(rc.state_machine.is(lksm::state::autonomous_activities));
}

// ! TODO: Refactor with composite SM & CoreTimer mock
TEST_F(RobotControllerTest, stateGameActivityStartedBackToMenu)
```